### PR TITLE
feat(cognito): support email based MFA

### LINKED ISSUE #31815: cognito: support email for multi-factor authentication (MFA)
### Describe the feature

Cognito now supports email for MFA.

Ref: https://aws.amazon.com/about-aws/whats-new/2024/09/amazon-cognito-user-pools-email-authentication-option/?nc1=h_ls

In CloudFormation, `EMAIL_OTP` was added to [EnabledMfas](https://docs.aws.amazon.com/ja_jp/AWSCloudFormation/latest/UserGuide/aws-resource-cognito-userpool.html#cfn-cognito-userpool-enabledmfas) property at [October 15, 2024](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/ReleaseHistory.html).

This options is not supported yet in L2 Construct.

### Use Case

To use email MFA

### Proposed Solution

Add email option to [MfaSecondFactor](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cognito.MfaSecondFactor.html).

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.162.0

### Environment details (OS name and version, etc.)

All

### PR DESCRIPTION
**Pull request by mazyu36**
### Issue # (if applicable)

Closes #31815.

### Reason for this change
To use email based MFA.

### Description of changes
Add email option to [MfaSecondFactor](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_cognito.MfaSecondFactor.html).

### Description of how you validated changes
Add unit tests and integ test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts [hunk 1]
```diff
@@ -1237,6 +1250,16 @@ export class UserPool extends UserPoolBase {
       }
       if (props.mfaSecondFactor!.otp) {
         enabledMfas.push('SOFTWARE_TOKEN_MFA');
+      } if (props.mfaSecondFactor!.email) {
+
+        if (props.email === undefined || this.emailConfiguration?.emailSendingAccount !== 'DEVELOPER') {
+          throw new Error('To enable email-based MFA, set `email` propety to the Amazon SES email-sending configuration.');
```
Comment (sujebane) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:1256]: Spelling mistake **property**

Comment (mazyu36) [reply packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts]: Thanks. fixed.

**Review by sujebane — APPROVED** [review]

File: packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts [hunk 2]
```diff
@@ -1237,6 +1250,16 @@ export class UserPool extends UserPoolBase {
       }
       if (props.mfaSecondFactor!.otp) {
         enabledMfas.push('SOFTWARE_TOKEN_MFA');
+      } if (props.mfaSecondFactor!.email) {
```
Comment (sumupitchayan) [inline packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:1253]: An idea to make this code more readable - we could add a helper function to validate email MFA, something like:

```ts
function validateEmailMfa(props: UserPoolProps) {
  if (!props.email || this.emailConfiguration?.emailSendingAccount !== 'DEVELOPER') {
    throw new Error('To enable email-based MFA, set `email` property to the Amazon SES email-sending configuration.');
  }
  
  if (props.advancedSecurityMode === AdvancedSecurityMode.OFF) {
    throw new Error('To enable email-based MFA, set `advancedSecurityMode` to `AdvancedSecurity.ENFORCED` or `AdvancedSecurity.AUDIT`.');
  }
}
```

and then replace the if statements with:

```ts
if (props.mfaSecondFactor!.otp) {
  enabledMfas.push('SOFTWARE_TOKEN_MFA');
}

if (props.mfaSecondFactor!.email) {
  validateEmailMfa(props);
  enabledMfas.push('EMAIL_OTP');
}
```

Makes it easier to follow this code

Comment (mazyu36) [reply packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:1253]: Thanks. I've updated. 
I thought private methods would be better so I went with that approach.

Comment (sumupitchayan) [reply packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts:1253]: Looks good, thanks @mazyu36 !

**Review by sumupitchayan — DISMISSED** [review]

**Comment by sumupitchayan** [comment]
@mergifyio update

**Review by sumupitchayan — APPROVED** [review]

**Pull request merged**